### ISSUE TITLE
Error: "Failed to create icon" on flutter pub run msix:create command

### ISSUE THREAD

**Issue by Abhishek0116** [YehudaKremer/msix#21]
I tried to build my sample application as windows installable file with this package but when I ran the final command `flutter pub run msix:create` then an error occurred.

I am using `msix: ^0.1.12`

Here are full error details or output from the terminal

```
C:\Users\Abhishek Kumar\Development\AndroidStudioProjects\TestProjects\my_test_project>flutter build windows

Building with unsound null safety
For more information see https://dart.dev/null-safety/unsound-null-safety

Building Windows application...

C:\Users\Abhishek Kumar\Development\AndroidStudioProjects\TestProjects\my_test_project>flutter pub run msix:create
getting config values..  [√]
validate config values..  [√]
cleaning temporary files..  [√]
create icons folder..  [√]
copy icons..  Unhandled exception:
fail to create icon C:/Users/Abhishek%20Kumar/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/msix-0.1.12/lib/assets/icons/icon.png: FileSystemException: Cannot copy file to 'C:\Users\Abhishek Kuma
r\Development\AndroidStudioProjects\TestProjects\my_test_project/build/windows/runner/Release/icons/icon.png', path = 'C:/Users/Abhishek%20Kumar/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/msi
x-0.1.12/lib/assets/icons/icon.png' (OS Error: The system cannot find the file specified.
, errno = 2)
#0      MsixFiles._copyIcon (package:msix/src/msixFiles.dart:271:7)
<asynchronous suspension>
#1      MsixFiles.copyIcons (package:msix/src/msixFiles.dart:38:33)
<asynchronous suspension>
#2      Msix.createMsix (package:msix/msix.dart:25:5)
<asynchronous suspension>
pub finished with exit code 255

C:\Users\Abhishek Kumar\Development\AndroidStudioProjects\TestProjects\my_test_project>
```

**Comment by YehudaKremer** [comment]
hello Abhishek

Maybe the problem is file path with a space,
I'm not sure yet.

Please try to setup the 3 icons in your  `pubspec.yaml` like that:

msix_config:
  logo_path: C:\<PathToIcon>\<Logo.png>
  start_menu_icon_path: C:\<PathToIcon>\<Icon.png>
  tile_icon_path: C:\<PathToIcon>\<Icon.png>

with that the problem is still happening?

**Comment by Abhishek0116** [comment]
@YehudaKremer Thanks for your reply.

I tried the above solution but now another error occurred

## Here's all command I ran

`flutter clean && flutter pub get`

`flutter build windows`

`flutter pub run msix:create`


## Here's the full output



```
C:\Users\Abhishek Kumar\Development\AndroidStudioProjects\TestProjects\beeta_lubricants>flutter clean && flutter pub get
Deleting .dart_tool...                                              18ms
Deleting .packages...                                                3ms
Deleting Generated.xcconfig...                                       2ms
Deleting flutter_export_environment.sh...                            2ms
Deleting ephemeral...                                               15ms
Deleting ephemeral...                                                3ms
Deleting ephemeral...                                                6ms
Deleting .flutter-plugins-dependencies...                            2ms
Deleting .flutter-plugins...                                         3ms
Running "flutter pub get" in beeta_lubricants...                    3.2s

C:\Users\Abhishek Kumar\Development\AndroidStudioProjects\TestProjects\beeta_lubricants>flutter build windows

Building with unsound null safety
For more information see https://dart.dev/null-safety/unsound-null-safety

Building Windows application...

C:\Users\Abhishek Kumar\Development\AndroidStudioProjects\TestProjects\beeta_lubricants>flutter pub run msix:create
getting config values..  [√]
validate config values..  [√]
cleaning temporary files..  [√]
create icons folder..  [√]
copy icons..  [√]
create manifest file..  [√]
copy VCLibs files..  Unhandled exception:
FileSystemException: Directory listing failed, path = 'C:/Users/Abhishek%20Kumar/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/msix-0.1.12/lib/assets/VCLibs/x64\*' (OS Error: The system cannot
 find the path specified.
, errno = 3)
Instance of '_StringStackTrace'pub finished with exit code 255

C:\Users\Abhishek Kumar\Development\AndroidStudioProjects\TestProjects\beeta_lubricants>

```

**Comment by micross** [comment]
the file path comes from the config file `.packages` which is uri encoded.

**Comment by Abhishek0116** [comment]
> the file path comes from the config file `.packages` which is uri encoded.

Hey, @micross thanks for this info.

What do I need to modify from my side? Can you please explain a bit?

**Comment by micross** [comment]
@Abhishek0116  this line [msix/main/lib/src/configuration.dart](https://github.com/YehudaKremer/msix/blob/94b80b33224d983025543c3495258fe8d543f402/lib/src/configuration.dart#L95)

the simple way is to remove the space in  `Abhishek Kumar`.

**Comment by YehudaKremer** [comment]
@micross Thank you for your inputs!

- i update the code to use `package_config.dart` instead of `.packages`.
- added Uri.decodeFull

@Abhishek0116 please check if the problem solved in version `0.1.14`

**Comment by YehudaKremer** [comment]
Hello @Abhishek0116 Did the fix help you?

**Comment by Abhishek0116** [comment]
Thanks, @micross and @YehudaKremer I have fixed my issue with your help and tested my created Windows app successfully. As @YehudaKremer said that issue is fixed in `version 0.1.14` and currently I tested successfully with `version 0.1.15` so I am closing this issue.
